src/factory/AnotherCloneFactory.sol

```diff
@@ -146,12 +149,16 @@ contract AnotherCloneFactory is AccessControlUpgradeable {
      *  Create new ERC721 collection
      *  Only the caller with role `PUBLISHER_ROLE` can perform this operation
      *
+     * @param _implementationId NFT implementation identifier
      * @param _name collection name
      * @param _salt bytes used for deterministic deployment
      */
-    function createCollection721(string memory _name, bytes32 _salt) external onlyRole(PUBLISHER_ROLE) {
+    function createCollection721(uint256 _implementationId, string memory _name, bytes32 _salt)
+        external
+        onlyRole(PUBLISHER_ROLE)
+    {
         // Create new NFT contract
-        ERC721AB newCollection = ERC721AB(Clones.cloneDeterministic(erc721Impl, _salt));
+        ERC721AB newCollection = ERC721AB(Clones.cloneDeterministic(erc721ImplAddresses[_implementationId], _salt));
```

Comment:
should we revert here if `erc721ImplAddresses[_implementationId]` is undefined or let `predictDeterministicAddress` do it?

Reply:
if we pass an undefined `_implementationId`, reading the array `erc721ImplAddresses` will revert (panic out of bond index). 

so in order to save gas, i did not add that check ( _implementationId >= erc721ImplAddresses.length)

the downside is that the revert message is a bit generic and not customized but i feel like its worth the few units of gas saved.